js/lib/DateFactory.js

```diff
@@ -259,4 +259,19 @@ DateFactory._dateToIlib = function(inDate, timezone, locale) {
     });
 };
 
+DateFactory._ilibToDate = function(ilibDate, timezone, locale) {
+    if (typeof(ilibDate) === 'undefined' || ilibDate === null) {
```

Comment:
ilibDate could be:

- a number (unix time)
- a string (ISO 8601 date format)
- an intrinsic Javascript Date object
- null, undefined, boolean, Symbol, array, or object other than an IDate
 
So this should catch all possibilities:

```
if (typeof(ilibDate) !== 'object' || !(ilibDate instanceof IDate)) {
    return ilibDate;
}
```